hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMDatanodeProtocolServer.java

```diff
@@ -280,6 +280,29 @@ public static SCMRegisteredResponseProto getRegisteredResponse(
     return cmd.getProtoBufMessage();
   }
 
+  private String constructCommandAuditMap(List<SCMCommandProto> cmds) {
+    StringBuilder auditMap = new StringBuilder();
+    auditMap.append("[");
+    for (SCMCommandProto cmd : cmds) {
+      if (cmd.getCommandType().equals(deleteBlocksCommand)) {
+        auditMap.append("commandType: ").append(cmd.getCommandType());
+        auditMap.append(" deleteTransactionsCount:");
+        auditMap.append(cmd.getDeleteBlocksCommandProto().getDeletedBlocksTransactionsList().size());
+      } else {
+        auditMap.append(cmd);
+      }
+      auditMap.append(" , ");
+    }
+    int len = auditMap.toString().length();
+    if (len > 3) {
+      auditMap.deleteCharAt(len - 1);
+      auditMap.deleteCharAt(len - 2);
+      auditMap.deleteCharAt(len - 3);
+    }
```

Comment:
I think this can be simplified.

(3 should be 2 if ` ,` is changed to `, `.)

```suggestion
    if (len > 3) {
      auditMap.delete(len - 3, len);
    }
```